### PR TITLE
Updated mitigations, simulation tuning

### PR DESCRIPTION
**Pull request by daeglin**

### CODE REVIEW

File: frontend/src/app/services/game.ts [hunk 1]
```diff
@@ -58,57 +70,93 @@ export class Game {
...
   get lastDate() {
     const lastState = last(this.simulation.modelStates);
-    return lastState?.date;
+    if (!lastState) throw new Error('Absent model state');
+
+    return lastState.date;
   }
 
   isFinished() {
-    const lastState = last(this.simulation.modelStates);
```
Comment (daeglin) [inline frontend/src/app/services/game.ts:65]: this was unused

Comment (daeglin) [inline frontend/src/app/services/game.ts:65]: This was unused

File: frontend/src/app/services/game.ts [hunk 2]
```diff
@@ -58,57 +70,93 @@ export class Game {
...
-    return Boolean(this.lastDate && this.lastDate >= this.endDate);
+    return this.lastDate >= this.endDate;
   }
```
Comment (daeglin) [inline frontend/src/app/services/game.ts:80]: lastDate is now always defined

File: frontend/src/app/services/game.ts [hunk 3]
```diff
@@ -58,57 +70,93 @@ export class Game {
...
+  }
+
+  calcMitigationEffect(mitigationState: MitigationState, forDate: string): MitigationEffect {
     let mult = 1.0;
     let cost = 0;
     let stabilityCost = 0;
+    let bordersClosed = false;
+    const vaccinationPerDay = this.vaccinationStartDate <= forDate ? this.vaccinationPerDay : 0;
+    const monthAsString = forDate.slice(5, 7);
+    const isSchoolBreak = monthAsString === '07' || monthAsString === '08';
```
Comment (jnwltr) [inline frontend/src/app/services/game.ts:103]: you can `forDate.slice(5, 7) in ['07', '08']`, although date fns would be a bit more robust

**Review by jnwltr — APPROVED** [review]

**Pull request merged**